.github/workflows/release.yml

```diff
@@ -11,24 +12,35 @@ on:
         description: 'Next SNAPSHOT version (without v). Defaults to current value'
         required: false
         type: string
+  pull_request:
 
 jobs:
   publish:
     runs-on: ubuntu-latest
 
     permissions:
+      pull-requests: write
       contents: write
+      security-events: write
+      packages: write
 
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
       - name: Set up JDK 17
         uses: actions/setup-java@v3
         with:
           java-version: '17'
           distribution: 'temurin'
 
+      - name: Fail if Release and not on Default branch
```

Comment:
Added this restriction as `IABTechLab/uid2-shared-actions/actions/commit_pr_and_merge` can't be called twice on feature branch (because base is changed). And the release shouldn't be created on feature branch.